File: ABC/264/d.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

string str;
vector<char> s;
int cnt = 0;

void swap(int num){
    char tmp = s[num];
    s[num] = s[num-1];
    s[num-1] = tmp;
}

int main(){
    cin>>str;
    for(int i=0;i<str.size();i++)s.push_back(str.at(i));
    // s = str;
    char t[] = {'a','t','c','o','d','e','r'};
    for(int i=0;i<7;i++){
        for(int j=0;j<s.size();j++){
            if(t[i]==s[j]){
                if(i==j)continue;
                while(i!=j){
                    swap(j);
                    j--;cnt++;
                }
            }
        }
    }
    cout<<cnt<<endl;
}

File: ABC/264/d.java
import java.util.*;

public class d {
    final static int MOD = 998244353;
    static char[] arr;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        arr = sc.next().toCharArray();
        sc.close();
        char[] str = "atcoder".toCharArray();
        long cnt = 0;
        for(int i=0;i<str.length;i++){
            for(int j=0;j<arr.length;j++){
                if(str[i]==arr[j]){
                    if(i==j)continue;
                    while(i!=j){
                        swap(j);
                        j--;
                        cnt++;
                    }
                }
            }
        }
        System.out.println(cnt);
    }
    static void swap(int num){
        char tmp = arr[num];
        arr[num] = arr[num-1];
        arr[num-1] = tmp;
    }
}
